lib/go/test/test.go

```diff
@@ -251,6 +251,56 @@ func assertEqual(t *testing.T, expected, actual interface{}) bool {
 	return assert.Fail(t, message)
 }
 
+func assertCadenceArrayEqual(t *testing.T, expected, actual []cadence.Value) bool {
+	assert.Len(t, actual, len(expected))
+
+	for _, resultVal := range actual {
+		found := false
+		for _, expectedVal := range expected {
+			if resultVal == expectedVal {
+				found = true
+			}
+		}
+
+		// One of the result values was not found in the expected list
+		if !assert.True(t, found) {
+			message := fmt.Sprintf(
+				"Arrays are not equal: \nexpected: %s\nactual  : %s",
+				expected,
+				actual,
+			)
+
+			return assert.Fail(t, message)
+		}
+	}
+	return true
+}
+
+func assertCadenceNodeDictionaryEqual(t *testing.T, expected, actual []cadence.KeyValuePair) bool {
```

Comment:
We're only evaluating equalities of keys because we're comparing `{String: Bool}` dictionaries and the values shouldn't matter because they should never be `false`, only `true` or `nil`, but I'll add value equality in as well just to be sure. and I'll move this and the array method into `staking_test_helpers` and add more comments since they are specific to that.